quartz/src/main/java/org/quartz/simpl/CascadingClassLoadHelper.java

```diff
@@ -107,9 +106,8 @@ public Class<?> loadClass(String name) throws ClassNotFoundException {
         Class<?> clazz = null;
         ClassLoadHelper loadHelper = null;
 
-        Iterator<ClassLoadHelper> iter = loadHelpers.iterator();
-        while (iter.hasNext()) {
-            loadHelper = iter.next();
+        for (ClassLoadHelper helper : loadHelpers) {
+            loadHelper = helper;
```

Comment:
Seems cleaner to do this assignment right before the `break`

Reply:
Umm I dont get it. The assignment is done just before the break

Reply:
More explicitly:
```java
for (ClassLoadHelper helper : loadHelpers) {
    try {
        clazz = helper.loadClass(name);
        loadHelper = helper;
        break;
    } catch (Throwable t) {
        throwable = t;
    }
}
```
but given the wider context you could go to:
```java
public Class<?> loadClass(String name) throws ClassNotFoundException {
    if (bestCandidate != null) {
        try {
            return bestCandidate.loadClass(name);
        } catch (Throwable t) {
            bestCandidate = null;
        }
    }

    Throwable throwable = null;

    for (ClassLoadHelper helper : loadHelpers) {
        try {
            Class<?> clazz = helper.loadClass(name);
            bestCandidate = helper;
            return clazz;
        } catch (Throwable t) {
            throwable = t;
        }
    }

    if (throwable instanceof ClassNotFoundException) {
        throw (ClassNotFoundException) throwable;
    } else {
        throw new ClassNotFoundException( String.format( "Unable to load class %s by any known loaders.", name), throwable);
    } 
}
```